### ISSUE TITLE
@template-extends should warn about unknown types used as parent type parameters

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1236]
https://getpsalm.org/r/e2203cd38f
```php
<?php
  
  /** 
   * @template T
   */
  class A {
     /** @return T */
     public function m() {}
  }

  /** 
   * @template TT
   * @template-extends A<Z>
   */
  class C extends A {}

  interface I {}
  /** @template-extends C<I> */
  class B extends C {}

  // atan((new B)->m());
```

**Expected**: unknown type Z
**Actual**: no issues

**Comment by weirdan** [comment]
Basically, type parameter list in `@template-extends` should consist of either known types or type parameters of the generic where `@template-extends` is seen, so that:
```php
/**
 * This is good
 * @template TT
 * @template-extends SomeParent<TT, RuntimeException>
 **/
/**
 * This is good as well
 * @template TT
 * @template-extends AnotherParent<RuntimeException>
 **/
/**
 * This is bad
 * @template TT
 * @template-extends SomeParent<SomeUnknownType>
 **/
```

**Comment by weirdan** [comment]
Also, it should not be possible to omit type parameters of the extended class:
```php
  /** 
   * @template T
   * @template V
   * @template U
   */
  class A {}

  /**
   * @extends A<int>
   */
  class CC extends A {}
```
This should complain about A expecting three type parameters, but only getting one in `@extends A<int>`.

**Comment by muglug** [comment]
Added checks for bad template-extends counts in cd0a744
